src/adapter/src/coord.rs

```diff
@@ -775,8 +802,9 @@ impl<S: Append + 'static> Coordinator<S> {
         }
 
         // Having installed all entries, creating all constraints, we can now relax read policies.
-        self.initialize_read_policies(&policies_to_set, DEFAULT_LOGICAL_COMPACTION_WINDOW_MS)
-            .await;
+        for (ts, policies) in policies_to_set {
+            self.initialize_read_policies(&policies, Some(ts)).await;
+        }
```

Comment:
It's somewhat inefficient to call this in a loop per timestamp. The underlying controller API can receive a single message with multiple IDs and timestamps. Though this is a failure of the read policy API, ideally `initialize_read_policies` would take some struct that contains `GlobalId`s and a timestamp per `GlobalId`.

It's probably fine for now since it looks like this will only ever iterate 2 times, but we probably are going to want to change this API to avoid slowing down start up.